### PR TITLE
Guide for Gradio ONNX model zoo on Huggingface

### PR DESCRIPTION
**Pull request by AK391**
# Description

added Guide for Gradio ONNX model zoo on Hugging Face

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import numpy as np
+import math
+import matplotlib.pyplot as plt
+import onnxruntime as rt
+import cv2
+import json
+import gradio as gr
+from huggingface_hub import hf_hub_download
+from onnx import hub
+import onnxruntime as rt
```
Comment (prasanthpul) [inline guides/Gradio_ONNX_on_Hugging_Face.md:68]: why is it imported twice?

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+Here we walk through setting up a example demo for EfficientNet-Lite4 using Gradio
+
+First we import our dependencies and download and load the efficientnet-lite4 model from the onnx model zoo. Then load the labels from the labels_map.txt file. We then setup our preprocessing functions, load the model for inference, and setup the inference function. Finally, the inference function is wrapped into a gradio inferface for a user to interact with. See the full code below.
+
+
+```python 
+import numpy as np
+import math
+import matplotlib.pyplot as plt
+import onnxruntime as rt
```
Comment (prasanthpul) [inline guides/Gradio_ONNX_on_Hugging_Face.md:62]: we like to use "as ort"

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+import cv2
+import json
+import gradio as gr
+from huggingface_hub import hf_hub_download
+from onnx import hub
+import onnxruntime as rt
+
+# lodas ONNX model from ONNX Model Zoo
```
Comment (prasanthpul) [inline guides/Gradio_ONNX_on_Hugging_Face.md:70]: "lodas" -> "loads"

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  return resultdic
+  
+title = "EfficientNet-Lite4"
+description = "EfficientNet-Lite 4 is the largest variant and most accurate of the set of EfficientNet-Lite model. It is an integer-only quantized model that produces the highest accuracy of all of the EfficientNet models. It achieves 80.4% ImageNet top-1 accuracy, while still running in real-time (e.g. 30ms/image) on a Pixel 4 CPU."
+examples = [['catonnx.jpg']]
+gr.Interface(inference, gr.inputs.Image(type="filepath"), "label", title=title, description=description, examples=examples).launch()
+```
+
+
+## Get started: How to contribute Gradio demos on HF spaces using ONNX models
```
Comment (prasanthpul) [inline guides/Gradio_ONNX_on_Hugging_Face.md:135]: should we add pointer to doc about how to create ONNX models?

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+ONNX Runtime inference can enable faster customer experiences and lower costs, supporting models from deep learning frameworks such as PyTorch and TensorFlow/Keras as well as classical machine learning libraries such as scikit-learn, LightGBM, XGBoost, etc. ONNX Runtime is compatible with different hardware, drivers, and operating systems, and provides optimal performance by leveraging hardware accelerators where applicable alongside graph optimizations and transforms. For more information please see the [official website](https://onnxruntime.ai/).
+
+## Setting up a Gradio Demo for EfficientNet-Lite4
+
+EfficientNet-Lite 4 is the largest variant and most accurate of the set of EfficientNet-Lite model. It is an integer-only quantized model that produces the highest accuracy of all of the EfficientNet models. It achieves 80.4% ImageNet top-1 accuracy, while still running in real-time (e.g. 30ms/image) on a Pixel 4 CPU. To learn more read the [model card](https://github.com/onnx/models/tree/main/vision/classification/efficientnet-lite4)
```
Comment (aliabd) [inline guides/Gradio_ONNX_on_Hugging_Face.md:51]: > ...of the set of EfficientNet-Lite *models.

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+Here we walk through setting up a example demo for EfficientNet-Lite4 using Gradio
+
+First we import our dependencies and download and load the efficientnet-lite4 model from the onnx model zoo. Then load the labels from the labels_map.txt file. We then setup our preprocessing functions, load the model for inference, and setup the inference function. Finally, the inference function is wrapped into a gradio inferface for a user to interact with. See the full code below.
```
Comment (aliabd) [inline guides/Gradio_ONNX_on_Hugging_Face.md:55]: Instead of one giant block with the full code i would recommend splitting into the different steps. 

show code for loading the labels, setting up preprocessing, loading the model ...etc

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 4 above)
Comment (aliabd) [inline guides/Gradio_ONNX_on_Hugging_Face.md:135]: Remove the Get started

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
+
+# Building Gradio Demos from the ONNX Model Zoo
```
[the remark this answers is not in the document]
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: thanks, updated

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 2 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md:61]: updated thanks

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 3 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: updated, thanks

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 4 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md:134]: this is specifically for adding pretrained models already on the onnx model zoo but we can add the link if users are also adding the model to the onnx zoo for the first time

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 1 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: updated thanks

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+## Setting up a Gradio Demo for EfficientNet-Lite4
+
+EfficientNet-Lite 4 is the largest variant and most accurate of the set of EfficientNet-Lite model. It is an integer-only quantized model that produces the highest accuracy of all of the EfficientNet models. It achieves 80.4% ImageNet top-1 accuracy, while still running in real-time (e.g. 30ms/image) on a Pixel 4 CPU. To learn more read the [model card](https://github.com/onnx/models/tree/main/vision/classification/efficientnet-lite4)
+
+Here we walk through setting up a example demo for EfficientNet-Lite4 using Gradio
+
+First we import our dependencies and download and load the efficientnet-lite4 model from the onnx model zoo. Then load the labels from the labels_map.txt file. We then setup our preprocessing functions, load the model for inference, and setup the inference function. Finally, the inference function is wrapped into a gradio inferface for a user to interact with. See the full code below.
+
+
+```python
```
[the remark this answers is not in the document]
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: sorry about that, updated

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+In this Guide, we'll walk you through:
+
+* Introduction of ONNX, ONNX model zoo, Gradio, and Hugging Face Spaces
+* How to setup a Gradio demo for EfficientNet-Lite4
+* How to contribute your own Gradio demos for the ONNX organization on Hugging Face
+
+See example EfficientNet-Lite4 Spaces demo to try below:
+
+<iframe src="https://hf.space/embed/onnx/EfficientNet-Lite4/+" frameBorder="0" height="450" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
```
[the remark this answers is not in the document]
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: sounds good updated

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+## Introduction
+
+In this Guide, we'll walk you through:
+
+* Introduction of ONNX, ONNX model zoo, Gradio, and Hugging Face Spaces
+* How to setup a Gradio demo for EfficientNet-Lite4
+* How to contribute your own Gradio demos for the ONNX organization on Hugging Face
+
+See example EfficientNet-Lite4 Spaces demo to try below:
```
[the remark this answers is not in the document]
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: updated

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+<iframe src="https://hf.space/embed/onnx/EfficientNet-Lite4/+" frameBorder="0" height="450" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
+
+## What is ONNX Model Zoo?
```
[the remark this answers is not in the document]
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: updated

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+Open Neural Network Exchange ([ONNX](https://onnx.ai/)) is an open standard format for representing machine learning models. ONNX is supported by a community of partners who have implemented it in many frameworks and tools. For example, if you have trained a model in TensorFlow, you can convert it to ONNX easily, and from there you could convert it to other frameworks like Pytorch, onnx runtime, Keras etc. For example, if you have trained a model in TensorFlow or PyTorch, you can convert it to ONNX easily, and from there run it on a variety of devices using an engine/compiler like ONNX Runtime.
```
[the remark this answers is not in the document]
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: updated

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+The [ONNX Model Zoo](https://github.com/onnx/models) is a collection of pre-trained, state-of-the-art models in the ONNX format contributed by community members. Accompanying each model are Jupyter notebooks for model training and running inference with the trained model. The notebooks are written in Python and include links to the training dataset as well as references to the original paper that describes the model architecture.
+
+
+## What is Hugging Face Spaces & Gradio?
+
+### Gradio
+
+Gradio lets users demo their machine learning models as a web app all in python code. Gradio wraps a python function into a user inferface and the demos can be launched inside jupyter notebooks, colab notebooks, as well as embedded in your own website and hosted on Hugging Face Spaces for free.
+
+for more info see: https://gradio.app/
```
[the remark this answers is not in the document]
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: updated

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 5 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: updated

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 4 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: updated

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 6 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md:54]: looking at past guides, the code to run the entire gradio app.py is in one code block so opted for that, I think we should keep that so users can copy paste the whole block to run a gradio demo if they want to

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (aliabd) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: no worries I actually learned about it from here

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 4 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: added a link to how to add models to onnx model zoo https://github.com/onnx/models/blob/main/.github/PULL_REQUEST_TEMPLATE.md

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 14]
```diff
@@ -0,0 +1,142 @@
+
+# Building Gradio Demos from the ONNX Model Zoo
+
+related_spaces: https://huggingface.co/spaces/onnx/EfficientNet-Lite4
+tags: ONNX, SPACES
```
[the remark this answers is not in the document]
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md:4]: Since the onnx team also helped with this we can just say gradio and onnx team?

[the remark this answers is not in the document]
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md:4]: added contribution by gradio and onnx team

**Comment by aliabd** [comment]
Space runs for a while then errors out 
Screen Shot 2022-04-27 at 6 15 21 AM

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 15]
```diff
@@ -0,0 +1,143 @@
+# Building_from_the_ONNX_Model_Zoo
+
+related_spaces: https://huggingface.co/spaces/onnx/EfficientNet-Lite4
+tags: ONNX, SPACES
+Contributed by Gradio and ONNX team
```
Comment (aliabd) [inline guides/Gradio_ONNX_on_Hugging_Face.md:5]: Contributed by Gradio and the [ONNX](https://onnx.ai/) team

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+
+## Introduction
+
+In this Guide, we'll walk you through:
+
+* Introduction of ONNX, ONNX model zoo, Gradio, and Hugging Face Spaces
+* How to setup a Gradio demo for EfficientNet-Lite4
+* How to contribute your own Gradio demos for the ONNX organization on Hugging Face
+
+Here's an example of an ONNX model: try out the EfficientNet-Lite4 demo below:
```
Comment (aliabd) [inline guides/Gradio_ONNX_on_Hugging_Face.md:16]: Here's an example of an ONNX model: try out the EfficientNet-Lite4 demo below.

(no colon in the end)

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+
+## What is the ONNX Model Zoo?
+Open Neural Network Exchange ([ONNX](https://onnx.ai/)) is an open standard format for representing machine learning models. ONNX is supported by a community of partners who have implemented it in many frameworks and tools. For example, if you have trained a model in TensorFlow or PyTorch, you can convert it to ONNX easily, and from there run it on a variety of devices using an engine/compiler like ONNX Runtime.
+
+The [ONNX Model Zoo](https://github.com/onnx/models) is a collection of pre-trained, state-of-the-art models in the ONNX format contributed by community members. Accompanying each model are Jupyter notebooks for model training and running inference with the trained model. The notebooks are written in Python and include links to the training dataset as well as references to the original paper that describes the model architecture.
+
+
+## What is Hugging Face Spaces & Gradio?
```
Comment (aliabd) [inline guides/Gradio_ONNX_on_Hugging_Face.md:26]: ## What *are Hugging Face Spaces & Gradio?

File: guides/Gradio_ONNX_on_Hugging_Face.md [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+title = "EfficientNet-Lite4"
+description = "EfficientNet-Lite 4 is the largest variant and most accurate of the set of EfficientNet-Lite model. It is an integer-only quantized model that produces the highest accuracy of all of the EfficientNet models. It achieves 80.4% ImageNet top-1 accuracy, while still running in real-time (e.g. 30ms/image) on a Pixel 4 CPU."
+examples = [['catonnx.jpg']]
+gr.Interface(inference, gr.inputs.Image(type="filepath"), "label", title=title, description=description, examples=examples).launch()
+```
+
+
+## How to contribute Gradio demos on HF spaces using ONNX models
+
+* add model to [onnx model zoo](https://github.com/onnx/models/blob/main/.github/PULL_REQUEST_TEMPLATE.md)
```
Comment (aliabd) [inline guides/Gradio_ONNX_on_Hugging_Face.md:137]: *Add model to *the onnx..

Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: thanks updated

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 17 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: thanks updated

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 16 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: thanks updated

File: guides/Gradio_ONNX_on_Hugging_Face.md (hunk 15 above)
Comment (AK391) [reply guides/Gradio_ONNX_on_Hugging_Face.md]: updated

**Comment by AK391** [comment]
> Space runs for a while then errors out Screen Shot 2022-04-27 at 6 15 21 AM

thanks it should be working now
Screen Shot 2022-04-27 at 5 15 19 AM
